### PR TITLE
fix-path-traversal-vulnerability

### PR DESCRIPTION
**Pull request by DharunKumar04**
This pull request addresses a path traversal vulnerability in the codebase, as reported by Snyk. The vulnerability arises from unvalidated user input being used as a path in the `fs.readFileSync` function, potentially allowing an attacker to read arbitrary files.

Changes Made

- Implemented input validation and sanitization to prevent path traversal attacks.
- Enforced strict file path handling to avoid unintended access to sensitive files.
- Enhanced security checks to safeguard against unauthorized access.

### CODE REVIEW

File: web/scripts/swagger-transform.js
Comment (tan-nhu) [inline web/scripts/swagger-transform.js:1]: Thanks, @DharunKumar04, for the PR. It seems a bit over-engineered.

Inputs for this script are specified in package.json (`"services": "restful-react import --config restful-react.config.js code"`). These inputs are internal, fixed, and not user-defined, reducing the risk of exposure.

If we really want to consider this a vulnerability, and the inputs in package.json are somehow used as an attack vector, we can simply hard-code `config` as `code` in this file. No need for the `transformModules` map, as it would also require hard-coding to map to `code` anyway.

```ts
const overridesFile = path.join('src/services', 'code', 'overrides.yaml')
const transformFile = path.join('src/services', 'code', 'transform.js')
```

Comment (DharunKumar04) [reply web/scripts/swagger-transform.js:1]: Thank you for your feedback, @tan-nhu . I appreciate your input on the pull request.

I understand your point about the inputs for this script being specified in package.json and the reduced risk of exposure. While the inputs are indeed internal and fixed, we wanted to ensure a robust and flexible solution that can adapt to potential changes in the future.

The use of the `transformModules` map makes the code more reusable. It enables us to add new configurations without modifying the main logic of the script, promoting code maintainability.

However, I appreciate your suggestion to simplify the code by hard-coding the '**code**' configuration directly in the script. If the consensus leans toward this simpler approach, we can certainly make that adjustment. Please feel free to provide your final thoughts on this, and we can proceed accordingly.

Once we have a clear direction, we can either continue with the current approach or implement the suggested change and close this PR.

Thank you for your valuable input and consideration.

Comment (tan-nhu) [reply web/scripts/swagger-transform.js:1]: @DharunKumar04 Let's change `config` to `code`. The change should be minimal. Currently Gitness UI connects to a single backend, the script was designed to support multiple ones. AFAIK, we won't have the second set of API in the short/medium term.

Also `restful-react` is being deprecated. We will adopt another library soon or later.

By the way, if you submit another PR. Please run `prettier` against the changed file(s). Thank you!

**Comment by DharunKumar04** [comment]
@tan-nhu  Thanks for your feedback and for considering the change. I've updated the **configuration as 'code'** as you suggested to simplify the script. This change aligns with the current usage pattern and reduces potential complexity.

Regarding the use of restful-react, we'll keep an eye on its deprecation and plan to transition to the new library when needed.and thanks for the input !!

I've also ensured that Prettier has been run against the changed files as requested.

If you have any further suggestions or concerns, please feel free to let me know. Your input is greatly appreciated!

**Review by tan-nhu — APPROVED** [review]

File: web/package.json
Comment (tan-nhu) [inline web/package.json:31]: Please remove `--config code`

**Review by tan-nhu — CHANGES_REQUESTED** [review]

File: web/package.json
Comment (DharunKumar04) [reply web/package.json]: Its done @tan-nhu
Waiting for your approval .

**Review by tan-nhu — APPROVED** [review]

**Comment by DharunKumar04** [comment]
Hi @hitesharinga, the PR has been approved by @tan-nhu. Please merge it so that we can close this. Thank you.

**Comment by hitesharinga** [comment]
Hey @DharunKumar04, Thank you for the Fix. I have merged it.

**Pull request merged**
